From b10d63c085509bd7800493a85e7bf2df5eb1fe0a Mon Sep 17 00:00:00 2001
From: Daiki Ueno <dueno@src.gnome.org>
Date: Thu, 23 Apr 2015 09:42:45 +0900
Subject: build: Avoid xgettext crash on MSYS2

isatty() on MSYS2 returns non-zero if the fd is stdout and is redirected
to /dev/null.  That lets xgettext to produce colorized output and leads
to an assertion failure during terminal type detection.

Although the problem should be fixed in MSYS2, isatty() could behave
wrongly in many ways on Windows.  Since gtk+ doesn't need colorized
output, it would be safer to bypass the terminal dependent code.

See also:
https://lists.gnu.org/archive/html/bug-gettext/2015-04/msg00004.html

https://bugzilla.gnome.org/show_bug.cgi?id=748346

diff --git a/configure.ac b/configure.ac
index ba8954c..005c678 100644
--- a/configure.ac
+++ b/configure.ac
@@ -668,7 +668,7 @@ ALL_LINGUAS="`grep -v '^#' "$srcdir/po/LINGUAS" | tr '\n' ' '`"
 AM_GLIB_GNU_GETTEXT
 
 AC_MSG_CHECKING([whether xgettext supports ui files])
-if $ac_cv_path_XGETTEXT --output=- --language=Glade $srcdir/gtk/ui/gtkfilechooserwidget.ui > /dev/null 2>&1; then
+if $ac_cv_path_XGETTEXT --output=- --color=no --language=Glade $srcdir/gtk/ui/gtkfilechooserwidget.ui > /dev/null 2>&1; then
   AC_MSG_RESULT([yes])
 else
   AC_MSG_RESULT([no])
-- 
cgit v0.10.2
